modules/certificate/certificate.service.js

```diff
@@ -2,22 +2,29 @@ const Certificate = require('./certificate.model');
 const RuleError = require('../../errors/rule.error');
 
 const {
-  STATUS_CODES: { NOT_FOUND, BAD_REQUEST },
+  STATUS_CODES: { NOT_FOUND, BAD_REQUEST, UNAUTHORIZED },
 } = require('../../consts/status-codes');
 const {
   CERTIFICATE_NOT_FOUND,
-  CERTIFICATE_ALREADY_EXISTS,
+  CERTIFICATE_NOT_VALID,
+  CERTIFICATE_HAVE_OWNER,
+  CERTIFICATE_IS_ACTIVE,
+  CERTIFICATE_IS_USED,
 } = require('../../error-messages/certificate.messages');
+const { USER_NOT_AUTHORIZED } = require('../../error-messages/user.messages');
 
 class CertificatesService {
   async getAllCertificates(skip, limit, user) {
     let filter;
 
     if (user.role === 'admin' || user.role === 'superadmin') {
       filter = {};
+    } else if (user.role === 'user') {
+      filter = { ownedBy: user.id };
     } else {
-      filter = { createdBy: user.id };
+      throw new RuleError(USER_NOT_AUTHORIZED, UNAUTHORIZED);
```

Comment:
useless because you have middleware for this

Reply:
Ok, will remove it, but need to debate a little more about it.  Added it for security reasons, because we don't know if new user roles will be added in the future.
Now permissions are getAllCertificates: and(isAuthorized, isUnlocked).

Reply:
done